backend/services/job.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * JobService that interacts with the job documents in the database
+ */
+import { CustomError, InternalError, ServiceError } from '../errors';
+import JobModel, { FIELDS_OWNER_PERMITTED_TO_UPDATE } from '../models/job';
+import { saveImage, deleteImage } from './image';
+
+export async function createJob(user, jobData, jobImages) {
+    console.debug('createJob service running')
+    const imageIds = [];
+
+    for (let image of jobImages) {
+        let imageId = await saveImage(image);
+        imageIds.push(imageId);
+    }
+
+    let job = null;
+    try {
+        job = await JobModel({
+            ...jobData, //TODO Sanitize input
+            client: user._id,
+            applicants: [],
+            imageIds: imageIds,
+        });
+    } catch (e) {
+        throw ServiceError.INVALID_JOB_RECEIVED.addContext(e.stack);
+    }
+    
+    //TODO Validate request fields like delievryDate, etc.
+
+    try {
+        return await job.save();
+    } catch (e) {
+        throw ServiceError.INVALID_JOB_RECEIVED.addContext(e.stack);
+    }
+}
+
+function filterUpdatePayload(payload, allowedFields) {
+    const filtered = {}  
+    allowedFields.forEach(field => {
```

Comment:
is there a reason we used forEach here instead of the other syntax [for (let element of arr)]

Reply:
No, I think forEach and for..of do the same thing and pretty sure they are equivalent performance wise. But I might change this forEach to for..of just to be consistent